File: ardoc/ardoc_mtime_diff_lstat.pl
#!/usr/bin/env perl
use Time::Local;

for (@ARGV) {
    @modtime = localtime((lstat($_))[9]);
    @curtime = localtime(time());
#    print "MMMM @modtime\n";
#    print "AMMM @modtime[0,1,2,3,4,5]\n"; 
    $timemod = timelocal(@modtime[0,1,2,3,4,5]);
    $timecur = timelocal(@curtime[0,1,2,3,4,5]);
    $timedif = $timecur - $timemod;
    $diffhour = $timedif/3600;
    $diftimprec_1 = int ($diffhour/24*10);  
#    print "DIFF1: $diffhour $diftimprec_1\n";
    $mtim=$modtime[7]+1;
    $mtimh=int ($modtime[2]/24*10);
    $ctim=$curtime[7]+1;
    $ctimh=int ($curtime[2]/24*10);
    $diftim=$ctim-$mtim;
    if($diftim < 0 ) {$diftim=$diftim + 366}
    $diftimprec=$diftim*10+$ctimh-$mtimh;
#    print "DIFF2: $diftim *10 + $ctimh - $mtimh $diftimprec\n"; 
    print "$diftimprec_1";
}
exit;

File: ardoc/ardoc_mtime_diff_lstat.py
#!/usr/bin/env python3
import os
import sys
import time

def main():
    for filepath in sys.argv[1:]:
        try:
            # Get the modification time using lstat
            mod_time_struct = time.localtime(os.lstat(filepath).st_mtime)
            
            # Get current time
            cur_time_struct = time.localtime()

            # Convert to seconds since epoch
            timemod = time.mktime(mod_time_struct)
            timecur = time.mktime(cur_time_struct)
            
            # Calculate difference
            timedif = timecur - timemod
            diffhour = timedif / 3600
            diftimprec_1 = int((diffhour / 24) * 10)
            
            print(diftimprec_1, end='')

        except FileNotFoundError:
            sys.stderr.write(f"Error: File not found: {filepath}\n")
        except Exception as e:
            sys.stderr.write(f"An error occurred: {e}\n")

if __name__ == "__main__":
    main()
